compiler-core/src/ast.rs

```diff
@@ -206,6 +206,7 @@ pub struct RecordConstructor<T> {
     pub name: EcoString,
     pub arguments: Vec<RecordConstructorArg<T>>,
     pub documentation: Option<EcoString>,
+    pub doc_position: Option<u32>,
```

Comment:
Do you think we should introduce a separate type to make it more semantically suggestive? I.e. something like `Spanned<EcoString>` (which is what Typst uses) instead of `(u32, EcoString)`.

Reply:
It turns out

> This should always be present if the documentation is present

doesn't appear to be true for this particular type due to the `put_doc` method below, which allows adding documentation outside of parsing. This appears to only be used in `build.rs`:

https://github.com/gleam-lang/gleam/blob/17fa7b8a218358e60685298d3cfb5af0a4d7f31c/compiler-core/src/build.rs#L263-L270

Not sure what's the best way forward here. Either we keep the field separate for this type, or we set the position to be the same as `location.start` in such a case.

Reply:
Actually, I found out that `build.rs` had enough information to provide the documentation position, so I changed `put_doc` to force a position to be specified alongside the documentation, and everything went well. Carry on!